		public void rollback(Savepoint savepoint)
			throws SQLException
		{
			checkAccess();
			try
			{
				m_delegate.rollback(savepoint);
			}
			catch(SQLException ex)
			{
				m_hadError = true;
				ThreadCategory.getInstance(getClass()).debug("setting bad flag [id=" + this + "]", ex);
				throw ex;
			}
		}

